xcode/Subconscious/Shared/Parsers/Subtext.swift

```diff
@@ -662,19 +705,11 @@ extension Subtext {
             .prefix(2) // Take first two blocks
         
         let output = validBlocks.joined(separator: "\n")
+            .truncate(maxLength: Self.maxExcerptSize)
         
         return output.isEmpty ? fallback : output
     }
     
-    func truncate(_ maxBlocks: Int = 2, fallback: String = "") -> Subtext {
-        // Filter out empty blocks
-        let validBlocks = blocks
-            .filter { block in !block.isEmpty }
-            .prefix(maxBlocks)
-        
-        return Subtext(base: self.base, blocks: Array(validBlocks))
-    }
-    
     static func excerpt(markup: String, fallback: String = "") -> String {
         let prefix = markup.truncate(maxLength: maxExcerptSize)
         return Subtext(markup: String(prefix)).excerpt(fallback: fallback)
```

Comment:
We should remove or update this function to use the logic from `truncate()`.

Reply:
I've reflowed the API surface there, now there's:

- `Subtext.truncate` (static) -> takes a string, truncates it at `Subtext.maxExcerptLength*maxBlocks` chars, parses it and takes `maxBlocks` from it
- `Subtext.truncate` (member) -> takes the first `maxBlocks` from this `Subtext`
- `Subtext.excerpt` (static) -> takes a string, calls `Subtext.truncate` (static) with `maxBlocks: Self.maxExcerptBlocks`
- `Subtext.excerpt` (member) -> calls `Subtext.truncate` (member) with `maxBlocks: Self.maxExcerptBlocks`

These make use of `String.truncateAtWordBounary`. I'm not sure what the implications of changing `String.truncate` everywhere are yet, will find out.